File: golang/16_limiter_factory/factory.go
package _6_limiter_factory

type RedisClient struct {
}

var GlobalFactory *Factory

func MustInit() error {
	//rc := GetRedisClient()
	f := &Factory{ /*rc*/ }

	GlobalFactory = f

	return nil
}

type Factory struct {
	rc *RedisClient
}

func (l *Factory) GenLimiter(useRedis bool) (Limiter, error) {
	if useRedis {
		return newRedisLimiter(l.rc)
	}

	return newMockLimiter()
}

File: myjava/src/main/java/org/example/limiter/Factory.java
package org.example.limiter;

public class Factory {
    private final RedisClient rc;

    public Factory() {
        this.rc = new RedisClient();
    }

    public Limiter genLimiter(boolean useRedis) {
        if (useRedis) {
            return new RedisLimiter(rc);
        }
        return new MockLimiter();
    }
}
